polaris-core/src/main/java/org/apache/polaris/core/config/BehaviorChangeConfigurationSince.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.config;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that should be applied to all BehaviorChangeConfiguration instances. The value here
+ * is the Polaris version when the flag was introduced. A spotless check should validate that
+ * behavior change flags are not present for more than 2 minor releases or 1 major release
+ */
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface BehaviorChangeConfigurationSince {
+  String value();
```

Comment:
maybe nitpicky, but how about `@BehaviorChangeConfiguration(added="1.0.0")`?

Perhaps `@BehaviorChangeConfigurationSince(added="1.0.0", expires="3.2.1")`?

Reply:
This is a cool idea; it would certainly make "extensions" easier. But I'm worried this takes the teeth out of the check; we want to enforce that these flags aren't around for a long time and letting the person making the flag choose an arbitrary expiration version seems to undermine that?

Reply:
`expires` is optional. Could be used to extent lifetime without tweaking global settings.

All-in-all, it comes down to code reviews, so I think the "teeth" are still there.

Reply:
That makes sense, I'm convinced. One small point to check, what do you propose to call the annotation now? "Behavior Change Configuration Since 1.0" seems idiomatic enough but "Behavior Change Configuration Since Added 1.0" seems a bit clunky.

Reply:
Changed to this for now. `Scope` seems to tread a little close to the `...Scoped` annotations in Quarkus, but maybe that's just me

Reply:
Maybe just `@BehaviorChange`?

Reply:
`@BehaviorChange` SGTM :+1: